openmc/mgxs/mgxs.py

```diff
@@ -2114,7 +2058,77 @@ def get_xs(self, in_groups='all', out_groups='all',
 
         return xs
 
-    def print_xs(self, subdomains='all', nuclides='all', xs_type='macro'):
+    def get_pandas_dataframe(self, groups='all', nuclides='all', moment='all',
+                             xs_type='macro', distribcell_paths=True):
+        """Build a Pandas DataFrame for the MGXS data.
+
+        This method leverages :meth:`openmc.Tally.get_pandas_dataframe`, but
+        renames the columns with terminology appropriate for cross section data.
+
+        Parameters
+        ----------
+        groups : Iterable of Integral or 'all'
+            Energy groups of interest. Defaults to 'all'.
+        nuclides : Iterable of str or 'all' or 'sum'
+            The nuclides of the cross-sections to include in the dataframe. This
+            may be a list of nuclide name strings (e.g., ['U-235', 'U-238']).
+            The special string 'all' will include the cross sections for all
+            nuclides in the spatial domain. The special string 'sum' will
+            include the cross sections summed over all nuclides. Defaults
+            to 'all'.
+        moment : int or 'all'
+            The scattering matrix moment to return. All moments will be
+             returned if the moment is 'all' (default); otherwise, a specific
+             moment will be returned.
+        xs_type: {'macro', 'micro'}
+            Return macro or micro cross section in units of cm^-1 or barns.
+            Defaults to 'macro'.
+        distribcell_paths : bool, optional
+            Construct columns for distribcell tally filters (default is True).
+            The geometric information in the Summary object is embedded into a
+            Multi-index column with a geometric "path" to each distribcell
+            instance.
+
+        Returns
+        -------
+        pandas.DataFrame
+            A Pandas DataFrame for the cross section data.
+
+        Raises
+        ------
+        ValueError
+            When this method is called before the multi-group cross section is
+            computed from tally data.
+
+        """
+
+        df = super(ScatterMatrixXS, self).get_pandas_dataframe(
+                groups, nuclides, xs_type, distribcell_paths)
+
+        # Add a moment column to dataframe
+        if self.legendre_order > 0:
+            # Insert a column corresponding to the Legendre moments
+            moments = ['P{}'.format(i) for i in range(self.legendre_order+1)]
+            moments = np.tile(moments, df.shape[0] / len(moments))
+            df['moment'] = moments
+
+            # Place the moment column before the mean column
+            mean_index = df.columns.get_loc('mean')
+            columns = df.columns.tolist()
+            df = df[columns[:mean_index] + ['moment'] + columns[mean_index:-2]]
```

Comment:
Using -1 would leave the "mean" column at the end. Instead, we want to
truncate it off the end and move it to the center of the DataFrame.
On May 13, 2016 2:13 PM, "Paul Romano" <notifications@github.com> wrote:

> In openmc/mgxs/mgxs.py
> <https://github.com/mit-crpg/openmc/pull/644#discussion_r63227759>:
>
> > +        """
> > +
> > +        df = super(ScatterMatrixXS, self).get_pandas_dataframe(
> > +                groups, nuclides, xs_type, distribcell_paths)
> > +
> > +        # Add a moment column to dataframe
> > +        if self.legendre_order > 0:
> > +            # Insert a column corresponding to the Legendre moments
> > +            moments = ['P{}'.format(i) for i in range(self.legendre_order+1)]
> > +            moments = np.tile(moments, df.shape[0] / len(moments))
> > +            df['moment'] = moments
> > +
> > +            # Place the moment column before the mean column
> > +            mean_index = df.columns.get_loc('mean')
> > +            columns = df.columns.tolist()
> > +            df = df[columns[:mean_index] + ['moment'] + columns[mean_index:-2]]
>
> Should that be -1 at the end there?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/openmc/pull/644/files/68d2e5b047d20964b866ce4fa555008f0ab3edb2#r63227759>
>